src/AWSMetadataUtilities.jl

```diff
@@ -0,0 +1,372 @@
+module AWSMetadataUtilities
+
+using .AWSExceptions
+using OrderedCollections: OrderedDict
+using HTTP
+using JSON
+
+include("AWSExceptions.jl")
+
+"""
+    _get_aws_sdk_js_files()
+
+Get a list of all AWS service API definition files from the `awsk-sdk-js` GitHub repository.
+
+# Returns
+- `Array{OrderedDict}`: Array of Dictionaries which contains information about each AWS Service
+"""
+function _get_aws_sdk_js_files()
+    headers = ["User-Agent" => "JuliaCloud/AWS.jl"]
+    url = "https://api.github.com/repos/aws/aws-sdk-js/contents/apis"
+
+    req = HTTP.get(url, headers)
+    files = JSON.parse(String(req.body), dicttype=OrderedDict)
+    filter!(f -> occursin(r".normal.json$", f["name"]), files)  # Only get ${Service}.normal.json files
```

Comment:
Doesn't matter much but may be clearer and faster:
```suggestion
    filter!(f -> endswith(f["name"], ".normal.json"), files)  # Only get ${Service}.normal.json files
```